src/shapesdemo/ShapeSubscriber.cpp

```diff
@@ -139,4 +147,11 @@ void ShapeSubscriber::on_requested_deadline_missed(
     m_mainWindow->addMessageToOutput(QString("Requested deadline missed"));
 }
 
+void ShapeSubscriber::on_liveliness_changed(
+        Subscriber*,
+        const LivelinessChangedStatus&)
+{
+    m_mainWindow->addMessageToOutput(QString("Liveliness changed"));
+}
```

Comment:
We should be more specific and say whether liveliness was lost or recovered. You can find that out by checking `LivelinessChangedStatus` passed to this method